controllers/territory_controller/tests.py

```diff
@@ -184,3 +187,85 @@ def test_BE_links(self) -> None:
             {StationCode.EY, StationCode.VB, StationCode.SZ, StationCode.PO},
             'Territory BE has incorrect territory links',
         )
+
+
+class TestTerritoryLockout(unittest.TestCase):
+    "Test that individual territories become 'locked' when claimed a set number of times"
+
+    _zone_0_territories = {StationCode.PN, StationCode.EY}
+    _zone_1_territories = {StationCode.YL, StationCode.PO}
+
+    def load_territory_owners(self, claim_log: ClaimLog) -> None:
+        # set territories owned by zone 0
+        for territory in self._zone_0_territories:
+            claim_log._station_statuses[territory] = Claimant.ZONE_0
+
+        # set territories owned by zone 1
+        for territory in self._zone_1_territories:
+            claim_log._station_statuses[territory] = Claimant.ZONE_1
+
+    @patch('controller.Supervisor.getFromDef')
+    @patch('territory_controller.get_robot_device')
+    def setUp(self, mock_get_robot: Mock, mock_getFromDef: Mock) -> None:
+        super().setUp()
+        claim_log = ClaimLog(record_arena_actions=False)
+        self.load_territory_owners(claim_log)
+        self.attached_territories = AttachedTerritories(claim_log)
+        with patch('territory_controller.TerritoryController.set_score_display'):
+            self.territory_controller = TerritoryController(
+                claim_log,
+                self.attached_territories,
+            )
+
+    @patch('controller.Supervisor.getFromDef')
+    def test_territory_lockout(self, mock_getFromDef: Mock) -> None:
+        """
+        Test a territory is locked after the correct number of claims and
+        disconnected territories aren't also locked
+        """
+        for i in range(LOCKED_OUT_AFTER_CLAIM):
+            # lock status is tested before capturing since the final
+            # iteration will lock the territory
+            for station in [StationCode.BE, StationCode.SZ]:  # assert BE, SZ unlocked
+                self.assertFalse(
+                    self.territory_controller._claim_log.is_locked(station),
+                    f"Territory {station.value} locked early on claim {i}",
+                )
+
+            # capture BE, SZ by opposing side
+            claimant = Claimant(i % 2)
+            for station in [StationCode.BE, StationCode.SZ]:
+                # when BE becomes locked SZ will be uncapturable
+                # which will prevent it from also locking
+                self.territory_controller.claim_territory(station, claimant, 0)
```

Comment:
Please can we make this explicit about which claimant rather than being clever about it.

Reply:
The idea of the loop is to alternate claims between teams since testing a team repeatedly claiming a territory is a different test. See 7eab2ec7cce443601227e50ac9594ca46c21fd7d

Reply:
(discussion happened, then https://github.com/srobo/competition-simulator/pull/320/commits/0c3292a16be1d0eeb2de0b44a9c353460a796bce)